### ISSUE TITLE
return type of function not detected in let

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#776]
```
$ clj-kondo --lint - <<< '(defn foo [] :foo) (let [a (foo)] (inc a))'
linting took 19ms, errors: 0, warnings: 0
```
